ppx/browser/ppx_deriving_json_runtime.ml

```diff
@@ -3,8 +3,23 @@ type t = Js.Json.t
 let to_json t = t
 let of_json t = t
 let to_string t = Js.Json.stringify t
-let of_string s = Js.Json.parseExn s
-let of_json_error msg = raise @@ Json.Decode.DecodeError msg
+
+exception Json_error of string
+
+let of_string s =
+  try Js.Json.parseExn s
+  with exn ->
+    let msg =
+      match Js.Exn.asJsExn exn with
+      | Some jsexn -> Js.Exn.message jsexn
+      | None -> None
+    in
+    let msg = Option.value msg ~default:"JSON error" in
```

Comment:
I would add a comment here, explaining that Js.Exn.message can't be none in the browser (and probably true in any js environment)

Reply:
done